bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnectionThread.java

```diff
@@ -0,0 +1,298 @@
+            if (!response.isOk()) {
+                throw new MPDException("could not authenticate");
+            }
+        }
+    }
+
+    private void openSocket() throws IOException {
+        logger.debug("opening connection to {} port {}", address, port);
+        Socket socket = new Socket(address, port);
+
+        inputStreamReader = new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8);
+        reader = new BufferedReader(inputStreamReader);
+        writer = new DataOutputStream(socket.getOutputStream());
+
+        this.socket = socket;
+    }
+
+    private void processPendingCommands() throws IOException {
+        MPDCommand currentCommand;
+
+        while (!disposed.get()) {
+            synchronized (pendingCommands) {
+                if (!pendingCommands.isEmpty()) {
+                    currentCommand = pendingCommands.remove(0);
+                } else {
+                    currentCommand = new MPDCommand("idle");
+                }
+
+                sendCommand(currentCommand);
+                if (currentCommand.getCommand().equals("idle")) {
+                    isInIdle.set(true);
+                }
+            }
+
+            MPDResponse response = readResponse(currentCommand);
+            if (!response.isOk()) {
+                insertCommand(new MPDCommand("clearerror"), 0);
+            }
+            listener.onResponse(response);
+        }
+    }
+
+    private void closeSocket() {
+        logger.debug("Closing socket");
+        BufferedReader reader = this.reader;
+        if (reader != null) {
+            try {
+                reader.close();
+            } catch (IOException e) {
+            }
+            this.reader = null;
+        }
+
+        InputStreamReader inputStreamReader = this.inputStreamReader;
+        if (inputStreamReader != null) {
+            try {
+                inputStreamReader.close();
+            } catch (IOException e) {
+            }
+            this.inputStreamReader = null;
+        }
+
+        DataOutputStream writer = this.writer;
+        if (writer != null) {
+            try {
+                writer.close();
+            } catch (IOException e) {
+            }
+            this.writer = null;
+        }
+
+        Socket socket = this.socket;
+        if (socket != null) {
+            try {
+                socket.close();
+            } catch (IOException e) {
+            }
+            this.socket = null;
+        }
+    }
+
+    private void sendCommand(MPDCommand command) throws IOException {
+        logger.trace("send command '{}'", command.asLine());
+        final DataOutputStream writer = this.writer;
+        if (writer != null) {
+            String line = command.asLine();
+            byte[] bytes = line.getBytes(StandardCharsets.UTF_8);
+            writer.write(bytes);
+            writer.write('\n');
+        } else {
+            throw new IOException("writer is null");
```

Comment:
This ends up in the UI if I read correctly what can the user do about this? Did it lose the connection. Please update the message.

Reply:
Please validate other messages when. might end up in the UI because of the behaviour at https://github.com/openhab/openhab-addons/pull/7870/files#diff-eac6da5ab58546ad9466910baded0b5fR86